odl/discr/discr_ops.py

```diff
@@ -499,9 +526,30 @@ def _resize_discr(discr, newshp, offset, discr_kwargs):
         else:
             new_maxpt.append(grid_max[axis] + (num_r + 0.5) * cell_size[axis])
 
-    return uniform_discr(new_minpt, new_maxpt, newshp, dtype=dtype, impl=impl,
-                         exponent=exponent, interp=interp, order=order,
-                         weighting=weighting, nodes_on_bdry=nodes_on_bdry)
+    fspace = FunctionSpace(IntervalProd(new_minpt, new_maxpt),
+                           out_dtype=dtype)
+    dspace = fn(np.prod(newshp), dtype=dtype, impl=impl, exponent=exponent,
+                weighting=weighting)
+
+    # Stack together the (unchanged) nonuniform axes and the (new) uniform
+    # axes in the right order
+    nonuni_axes = [i for i in range(discr.ndim)
```

Comment:
I would prefer if this was moved down, to the `if` statemnt further down, that way we don't have conditionals in two places and the code becomes less complex.